### ISSUE TITLE
TTY detection complicates use in scripts

### ISSUE THREAD

**Issue by dvogel** [sbdchd/squawk#265]
Since processes inherit the stdin of their parent process, squawk's consumption of stdin regardless of the `path` argument can lead to surprising behavior when scripting it:

```
find src/main/resources/db/migration/ -type f -iname '*.sql' \
    | while read -r filename; do
        squawk "$filename"
    done
```

This will complain that the query could not be parsed even if this command parses it fine:

```
squawk "$filename"
```

This is because in the `while` loop `squawk` inherits the `find` output as it's stdin and ignores the contents of the file named on the command line. I would expect stdin to be consumed if and only if a path is not named on the command line. If the tty detection approach is necessary for some other use case, it would be nice if squawk provided and error complaining of a path being named when stdin is not a tty.

I understand that the repro case above can be accomplished like so:

```
squawk src/main/resources/db/migration/*.sql
```

However the full use case here involves some filtering of the file list:

```
find src/main/resources/db/migration/ -type f -iname '*.sql' \
    | sort -h \
    | grep --line-regexp --invert-match -f squawk-excludes \
    | grep --line-regexp -f <(git diff --name-only "$gitrev") \
    | while read -r filename; do
        squawk --pg-version=13.0 "$filename"
    done
```

For now I've worked around this by overriding the stdin inheritence:

```
while read -r filename; do
    squawk --pg-version=13.0 "$filename" < /dev/stdin
done
```

**Comment by chdsbd** [comment]
Thanks for the report @dvogel

This is definitely a bug that should be fixed to match the behavior you've outlined

**Comment by chdsbd** [comment]
This should be fixed in the latest version, 0.23.0
